hbase-http/src/main/java/org/apache/hadoop/hbase/http/HttpServer.java

```diff
@@ -1129,7 +1137,7 @@ public void setThreads(int min, int max) {
   private void initSpnego(Configuration conf, String hostName, String usernameConfKey,
     String keytabConfKey, String kerberosNameRuleKey, String signatureSecretKeyFileKey)
     throws IOException {
-    Map<String, String> params = new HashMap<>();
+    Map<String, String> params = conf.getPropsWithPrefix(HTTP_SPNEGO_AUTHENTICATION_PREFIX);
```

Comment:
What is the impact of this change to the code snippets calling params.put in lines that follow?

Reply:
```java
params.put(AuthenticationFilter.AUTH_TYPE, "kerberos");
```

On `initSpnego()`, hbase will use `KerberosAuthenticationHandler`.
And in order to use `.endpoint.whitelist` feature from the class, `hbase.security.authentication.spnego.kerberos.endpoint.whitelist` param should be added.

- https://github.com/apache/hadoop/blob/rel/release-3.3.4/hadoop-common-project/hadoop-auth/src/main/java/org/apache/hadoop/security/authentication/server/KerberosAuthenticationHandler.java#L104

I think adding parameters which starts with `hbase.security.authentication.spnego.` more makes sense like the way on the hadoop side:

- https://github.com/apache/hadoop/blob/rel/release-3.3.4/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/AuthenticationFilterInitializer.java#L72
- https://github.com/apache/hadoop/commit/67e02a92e0b9c4da3dcdd01f231a98f243f12d06#diff-4e9d7dccc4530205e71b54fe7f967135aeca170cff5ace98b5b7f04304153813R503